Scan.cpp

```diff
@@ -0,0 +1,99 @@
+#include "ProcessManager.h"
+
+#include <assert.h>
+
+#include <vector>
+#include <utility>
+
+using namespace std;
+
+/**
+ * Takes a maps file and returns a vector of pairs 
+ * having the beginning and ending addresses of readable
+ * memory locations.
+ * 
+ * @param fileBuffer the maps file for a process
+ * @returns a vector pairs
+ */
+vector<pair<long, long>> getRanges(char *fileBuffer){
+	vector<pair<long, long>> ranges;
+	char *p;
+	char *m = fileBuffer;
+	while((p=strstr(m, "\n"))!=NULL){
+		char *start = p-1;
+		while( (*start) != '\n' && start>=fileBuffer)
+		{
+			start--;
+		}
+		start++;
+		char *c=strstr(start, "---p");
+		if(c&&c<strstr(start,"\n")){
+			m=p+1;
+			continue;
+		}
+		char chAddress[1024];
+		assert(chAddress != NULL);
+		memset(chAddress, 0, 1024);
+		for(int i=0;*start!='-';i++)
+		{
+			chAddress[i] = *start;
+			start++;
+		}
+		pair<long, long> range;
+		range.first = strtol(chAddress, NULL, 16);
+		memset(chAddress, 0, 1024);
+		start++;
+		for (int i = 0; *start != ' '; i++)
+		{
+			chAddress[i] = *start;
+			start++;
+		}
+		range.second = strtol(chAddress, NULL, 16);
+		ranges.push_back(range);
+		m = p + 1;
+	}
+	return ranges;
+}
+
+/**
+ * Takes in a target value and matches it against all
+ * readable memory locations' data.
+ * 
+ * @param target the value to match
+ * @returns vector of addresses
+ */
+vector<long> ProcessManager::scanProcess(long target){
```

Comment:
add ability to search from a vector if a vector is provided. One way to do is that if No vector is provided to the function, it will do the getRanges thing else it will take up addresses from the vector provided

Reply:
I'd like to work on that in a separate PR, once this is merged. I hope that's fine.